test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -167,24 +167,42 @@ call_log_manager::call_log_query_timestamp(const json &call_log_entry)
     /* Convert the config char * to a string object. */
     std::string config = call_log_entry["input"]["config"].get<std::string>();
 
-    /*
-     * A generated call log without a configuration string in the set timestamp entry will have the
-     * string "(null)", default to all_durable.
-     */
-    if (config == "(null)")
-        config = "get=all_durable";
-
+    std::string class_name = call_log_entry["class_name"].get<std::string>();
     std::string hex_ts;
     bool ts_supported;
+    int ret;
+    /* Check whether we're querying a global connection timestamp or a session transaction
+     * timestamp. */
+    if (class_name == "connection") {
+        /*
+         * A generated call log without a configuration string in the set timestamp entry will have
+         * the string "(null)", default to all_durable.
+         */
+        if (config == "(null)")
+            config = "get=all_durable";
+
+        ret = _conn->query_timestamp(config, hex_ts, ts_supported);
+    } else if (class_name == "session") {
+        /*
+         * A generated call log without a configuration string in the set timestamp entry will have
+         * the string "(null)", default to read timestamp.
+         */
+        if (config == "(null)")
+            config = "get=read";
+
+        const std::string session_id = call_log_entry["session_id"].get<std::string>();
+        session_simulator *session = get_session(session_id);
+
+        ret = session->query_timestamp(config, hex_ts, ts_supported);
+    }
 
-    int ret = _conn->query_timestamp(config, hex_ts, ts_supported);
     int ret_expected = call_log_entry["return"]["return_val"].get<int>();
     /* The ret value should be equal to the expected ret value. */
     assert(ret == ret_expected);
 
     if (ret != 0)
```

Comment:
Do we want to throw an exception or an assert would do the job?

Reply:
I was following the style of the previous functions for this.